.werft/installer-tests.ts

```diff
@@ -41,14 +116,22 @@ const INFRA_PHASES: { [name: string]: InfraConfig } = {
         makeTarget: "managed-dns",
         description: "Sets up external-dns & cloudDNS config",
     },
+    GENERATE_KOTS_CONFIG: {
+        phase: "generate-kots-config",
+        makeTarget: `generate-kots-config storage=${randomize("storage", cloud)} registry=${randomize(
```

Comment:
Suggestion: performing randomization in this location means that loading this file isn't an idempotent operation. If a randomized test configuration fails we'll likely need to repeat that test case (either to rule out a transient failure or reproduce the failure). Because these settings are encoded during the file evaluation we don't have a means of injecting an exact configuration.

Phase parameterization can be kept out of scope for this PR (espcially because it'll take time to define how we configure phases), but we should definitely follow up on this in another issue/PR for Q3. If we defer to Q3 I'd love to capture this requirement in a ticket before we merge this PR.

**EDIT**: Leaving the above comment as it's still relevant, but less so.

In `install/tests/Makefile` we do provide mechanisms for setting the exact configuration so we should be able to reproduce a given test case if we omit werft. This reduces the challenge of randomizing these values 🎉  , but we'll still need to account for this in the future.